lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -500,6 +500,11 @@ static bool createsInfiniteSpecializationLoop(ApplySite Apply) {
 
 static bool shouldNotSpecialize(SILFunction *Callee, SILFunction *Caller,
                                 SubstitutionMap Subs = {}) {
+  // Ignore "do not specialize" markers in embedded Swift -- specialization is
+  // mandatory.
+  if (Callee->getModule().getOptions().EmbeddedSwift)
```

Comment:
Would be nice if we could do this for perf annotations too, but that's probably a separate patch.

Reply:
Hm, I wonder if there is actually anything to fix for perf annotations: (1) The stdlib only has a handful of functions with the "do not specialize" marker and they seem to mostly be slow paths that do cause allocations (and thus are not going to be usable in `@noLocks` or `@noAllocations` code anyway). (2) Non-stdlib code probably shouldn't use the "do not specialize" marker as it's an underscored attribute, and arguably it could be considered to be the user's problem if they use the marker and then try to use such a code in a perf-annotated function...